internal/direct.go

```diff
@@ -55,17 +62,50 @@ func (reporter directReporter) Report(format string, pointLines string) (*http.R
 
 	apiURL := reporter.serverURL + reportEndpoint
 	req, err := http.NewRequest("POST", apiURL, &buf)
-	req.Header.Set(contentType, octetStream)
-	req.Header.Set(contentEncoding, gzipFormat)
-	req.Header.Set(authzHeader, bearer+reporter.token)
 	if err != nil {
 		return &http.Response{}, err
 	}
 
+	req.Header.Set(contentType, octetStream)
+	req.Header.Set(contentEncoding, gzipFormat)
+	req.Header.Set(authzHeader, bearer+reporter.token)
+
 	q := req.URL.Query()
 	q.Add(formatKey, format)
 	req.URL.RawQuery = q.Encode()
 
+	return execute(req)
+}
+
+type directEventReporter struct {
+	serverURL string
+	token     string
+}
+
+// NewDirectEventReporter create a event Reporter
+func NewDirectEventReporter(server string, token string) Reporter {
+	return &directEventReporter{serverURL: server, token: token}
+}
+
+func (reporter directEventReporter) Report(format string, event string) (*http.Response, error) {
```

Comment:
Could this just be moved as `ReportEvent` within the `directReporter` struct?

Reply:
ok

Reply:
@laullon Previous comment is still not addressed. Let's remove the `directEventReporter` struct and just add a `ReportEvent` method to the existing `Reporter` interface/implementation. 

Basically the `Reporter` interface is a thing wrapper around the WF API and will support the `Report` (/report) and `ReportEvent` (/api/v2/event) endpoints.

Reply:
this change will add a lot of if's that we removed on the previous comment...